src/MessageFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RabbitEvents\Event;
+
+use Illuminate\Contracts\Container\Container;
+use RabbitEvents\Foundation\Support\Payload;
+use RabbitEvents\Foundation\Contracts\Transport;
+use RabbitEvents\Foundation\Message;
+
+class MessageFactory
+{
+    public function __construct(private Transport $transport)
+    {
+    }
+
+    /**
+     * @param ShouldPublish $event
```

Comment:
Notice: I'm not sure about adding phpdoc like that because all data already typed